rr/gstreamer/gst_media.py

```diff
@@ -128,19 +129,61 @@ def _on_new_buffer(self, appsink, data):
 
         buf = sample.get_buffer()
 
-        gst_memory = buf.get_all_memory()
-        ret, minfo = gst_memory.map(gst.MapFlags.READ)
+        gst_memory_obj = buf.get_all_memory()
+        ret, minfo = gst_memory_obj.map(gst.MapFlags.READ)
 
         if minfo.data is None:
             return gst.FlowReturn.ERROR
 
-        self.callback(minfo.data)
+        caps = sample.get_caps()
+        width, height, format = (caps.get_structure(0).get_value("width"),
+                                 caps.get_structure(0).get_value("height"),
+                                 caps.get_structure(0).get_value("format")
+                                 )
```

Comment:
Since you are already passing the sample to `GstImage`'s constructor, why not move all of this logic there?

Reply:
I'm following the specification to pass these parameters: width, height, format, through the constructor; that's is how it is supposed to be done for other parts of code on the design.

Reply:
And:
```py
        gst_memory_obj = buf.get_all_memory()
        ret, minfo = gst_memory_obj.map(gst.MapFlags.READ)
```
?

Reply:
Same, we want to keep the mapping over there for the moment.

Reply:
Thanks, I added the mapping into the class constructor.